http/adapter.go

```diff
@@ -44,5 +45,7 @@ func (as *adapterService) Publish(ctx context.Context, token string, msg broker.
 	}
 	msg.Publisher = thid.GetValue()
 
+	msg.Created = time.Now().Unix()
```

Comment:
I think that this field should be created when we create a Message. In this case, that would be [here](https://github.com/mainflux/mainflux/blob/57f7ee25950548aa4a1e4cfddd35db2f7e5ed051/http/api/transport.go#L112).
```go
	msg := broker.Message{
		Protocol:    protocol,
		ContentType: ct,
		Channel:     chanID,
		Subtopic:    subtopic,
                Created:     time.Now().Unix(),
		Payload:     payload,
	}
```

Reply:
I think that the message creation should be done in service layer like it's done in most of the services.

Reply:
We set publisher here because it's not available before. If we can set fields at the time of creation, I don't see any reason not to do so. We're doing that for all the other fields.

Reply:
@drasko @nmarcetic what do you think about?

Reply:
I don't have strong preferences, really. Maybe I agree with @dusanb94 more, I mean why not add it in the time of creation if we can, it's just easier. I don't understand @manuio 
`I think that the message creation should be done in service layer like it's done in most of the services` Its not like that now, you want to remove message creation from transport and put it in service?

Reply:
@nmarcetic Yes I think that `broker.Message{}` should be created in the service layer and that all the data should be passed using `publishReq` (Currently we create the message and pass it as publishReq parameter). Also it's consistent with MQTT, Lora and OPC-UA where there is no transport layer for messaging.

Reply:
@manuio I understand now. I agree with you, but :) All this data that makes message actually is decoded on transport layer here (it's go-kit specific, can't compare with other no go-kit services) and this data is made from things like request body, request URL etc... everything we have in the transport layer. If we do propagation of request to service, it's not really clear to me. I really do agree with you, but think this is really specific and its to messy to align it with other no go-kit services or just not HTTP oriented.